pep-0544.txt

```diff
@@ -842,6 +904,43 @@ of simplicity, we propose to not support optional methods or attributes.
 We can always revisit this later if there is an actual need.
 
 
+Allow only protocol methods and force use of getters and setters
+----------------------------------------------------------------
+
+One could argue that protocols typically only define methods, but not
+variables. However, using getters and setters in cases where only a
+simple variable is needed would be quite unpythonic. Moreover, the widespread
+use of properties (that are often act as validators) in large code bases
+is partially due to previous absence of static type checkers for Python,
+the problem that PEP 484 and this PEP are aiming to solve.
+
+
+Support non-protocol members
+----------------------------
+
+There was an idea to make some methods "non-protocol" (i.e. not necessary
+to implement, and inherited in explicit subclassing), but it was rejected,
+since this complicates things. For example, consider this function::
+
+  def fun(m: Mapping):
+      m.keys()
+
+The question is should this be an error? We think most people would expect
```

Comment:
Honestly I don't understand why this is even a question. OF COURSE it is valid, keys() is part of the Mapping class/protocol/ABC.

I do think I understand what's going on here, you're considering that keys() is a non-abstract method in Mapping, and the question is whether non-abstract methods are considered part of the protocol by default. I just think the example is a poor one (unless one happens to know that keys() has a default implementation in the Mapping class). So I think you just have to switch to an example that doesn't rely on a standard ABC.